pubsub.go

```diff
@@ -162,33 +182,47 @@ func (p *PubsubValueStore) Subscribe(key string) error {
 		return cmp > 0 || cmp == 0 && src == myID
 	})
 
-	sub, err := p.ps.Subscribe(topic)
-	if err != nil {
+	if err := p.fillTopicInfo(ti, topic); err != nil {
```

Comment:
We're not holding the lock.

Reply:
Might as well to be safe. I think I was relying on p.ps.Join only succeeding the first time it's called (i.e. only 1 Topic can exist per topic). This is probably not something we should overly rely on though.

Reply:
Got it. Won't rebroadcast concurrently read this?

Reply:
probably, better off protecting this anyway.

Reply:
better?

Reply:
Note: 

> probably, better off protecting this either way.

This is _never_ a good idea. If you don't know whether or not a lock is necessary, the code is probably buggy.